beacon-chain/sync/validate_aggregate_proof.go

```diff
@@ -44,32 +44,39 @@ func (r *Service) validateAggregateAndProof(ctx context.Context, pid peer.ID, ms
 		traceutil.AnnotateError(span, err)
 		return false
 	}
-	m, ok := raw.(*ethpb.AggregateAttestationAndProof)
+	m, ok := raw.(*ethpb.SignedAggregateAttestationAndProof)
 	if !ok {
 		return false
 	}
 
+	// Verify this is the first aggregate received from the aggregator with index and slot.
+	if r.hasSeenAggregatorIndexSlot(m.Message.Aggregate.Data.Slot, m.Message.AggregatorIndex) {
```

Comment:
Can you check these are set? Looks like a prime target for nil pointer panic

Reply:
sounds good